internal/util/flowgraph/node.go

```diff
@@ -53,26 +53,156 @@ type BaseNode struct {
 	maxParallelism int32
 }
 
+// manage nodeCtx
+type nodeCtxManager struct {
+	inputNodeCtx *nodeCtx
+	closeWg      *sync.WaitGroup
+	checker      *timerecord.GroupChecker
+
+	closeOnce sync.Once
+
+	inputNodeCloseCh chan struct{} // notify input node work to exit
+	ddNodeCloseCh    chan struct{} // notify ddnode and downstream node work to exit
+}
+
+// NewNodeCtxManager init with the inputNode and fg.closeWg
+func NewNodeCtxManager(nodeCtx *nodeCtx, closeWg *sync.WaitGroup) *nodeCtxManager {
+	return &nodeCtxManager{
+		inputNodeCtx:     nodeCtx,
+		closeWg:          closeWg,
+		inputNodeCloseCh: make(chan struct{}),
+		ddNodeCloseCh:    make(chan struct{}),
+	}
+}
+
+// Start invoke Node `Start` method and start a worker goroutine
+func (nodeCtxManager *nodeCtxManager) Start(checker *timerecord.GroupChecker) {
+	// in dmInputNode, message from mq to channel, alloc goroutines
+	// limit the goroutines in other node to prevent huge goroutines numbers
+	nodeCtxManager.closeWg.Add(2)
+	go nodeCtxManager.inputNodeWork(checker)
+	go nodeCtxManager.ddNodeWork(checker)
+}
+
+func (nodeCtxManager *nodeCtxManager) inputNodeWork(checker *timerecord.GroupChecker) {
+	defer nodeCtxManager.closeWg.Done()
+	inputNode := nodeCtxManager.inputNodeCtx
+	for {
+		select {
+		case <-nodeCtxManager.inputNodeCloseCh:
+			// remove input node checker
+			checker.Remove(fmt.Sprintf("nodeCtxTtChecker-%s", inputNode.node.Name()))
```

Comment:
Also spelling the name is not a good idea, add a const string equals to the prefix, or remove the prefix